### PR TITLE
Flowauth token validation

### LINKED ISSUE #799: Client side validation for new token page
Subcategory of #54

Added client-side validation and appropriate error messages for new token page for admin users.

### PR DESCRIPTION
**Pull request by BhavinPanch**
Closes #799 

### I have:

- [ ] Formatted any Python files with [black](https://github.com/ambv/black)
- [x] Brought the branch up to date with master
- [ ] Added any relevant Github labels
- [x] Added tests for any new additions
- [ ] Added or updated any relevant documentation
- [ ] Added an Architectural Decision Record (ADR), if appropriate
- [x] Added an [MPLv2 License Header](https://www.mozilla.org/en-US/MPL/headers/) if appropriate
- [x] Updated the [Changelog](https://github.com/Flowminder/FlowKit/blob/master/CHANGELOG.md) 

### Description

Added client-side validation and helpful error messages on new token page for user.

### CODE REVIEW

File: flowauth/frontend/cypress/integration/add_new_token.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            force: true
+        });
+        cy.get("#name")
+            .clear({
+                force: true
+            });
+        //checking validation text
+        cy.get("#name-helper-text").should(
+            "have.text",
+            "Token name can not be blank."
```
Comment (greenape) [inline flowauth/frontend/cypress/integration/add_new_token.js]:
```suggestion
            "Token name cannot be blank."
```

File: flowauth/frontend/src/TokenDetails.js [hunk 2]
```diff
@@ -67,9 +71,23 @@ class TokenDetails extends React.Component {
...
-    });
+    var letters = /^[A-Za-z0-9_]+$/;
+    let name = event.target.value;
+    if (name.match(letters)) {
+      this.setState(Object.assign(this.state, {
+        name_helper_text: ""
+      }));
+    } else if (name.length == 0) {
+      this.setState(Object.assign(this.state, {
+        name_helper_text: "Token name can not be blank."
```
Comment (greenape) [inline flowauth/frontend/src/TokenDetails.js]:
```suggestion
        name_helper_text: "Token name cannot be blank."
```

File: flowauth/frontend/cypress/integration/add_new_token.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        cy.contains("#name-helper-text").should("not.exist");
+    });
+    it("Add new token", function () {
+        cy.get("#new").click();
+        //Add new token 
+        cy.get("#name").type("TOKEN_TEST01", {
+            force: true
+        });
+        cy.contains("Save").click();
+        cy.wait(500);
```
Comment (greenape) [inline flowauth/frontend/cypress/integration/add_new_token.js]: Is the wait necessary?

File: flowauth/frontend/cypress/integration/add_new_token.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        // Log in and navigate to user details screen
+        cy.login();
+        cy.visit("/");
+        cy.get("#servers").click();
+    });
+    it("Add token name with space", function () {
+        cy.get("#new").click();
+        // adding username with space
+        cy.get("#name").type("Token ", {
+            force: true
```
Comment (greenape) [inline flowauth/frontend/cypress/integration/add_new_token.js]: Can you add a comment explaining why this is required?

File: flowauth/frontend/cypress/integration/add_new_token.js (hunk 3 above)
Comment (BhavinPanch) [reply flowauth/frontend/cypress/integration/add_new_token.js]: I have removed it. It was for delaying next step but I tested it again and it's not required.

**Pull request merged**